aggregator/src/main/scala/ai/chronon/aggregator/windowing/SawtoothOnlineAggregator.scala

```diff
@@ -89,10 +89,6 @@ class SawtoothOnlineAggregator(val batchEndTs: Long,
     val batchIr = Option(finalBatchIr).getOrElse(normalizeBatchIr(init))
     val headRows = Option(streamingRows).getOrElse(Array.empty[Row].iterator)
 
-    if (batchEndTs > queryTs) {
```

Comment:
If we get here, we will instead throw an NPE right? Let's keep this around. Your other changes will gate the control from throwing this anyways.

Reply:
sounds good.